src/NuGet.Clients/NuGet.PackageManagement.UI/Models/LicenseFileText.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel;
+using Microsoft.VisualStudio.Threading;
+using System.Threading.Tasks;
+using NuGet.VisualStudio;
+using System.Globalization;
+using System;
+
+namespace NuGet.PackageManagement.UI
+{
+    internal class LicenseFileText : IText, INotifyPropertyChanged
+    {
+        private string _text;
+        private string _licenseText;
+        private string _licenseHeader;
+        private readonly string _licenseFileLocation;
+        private Func<string, Task<string>> _loadFileFromPackage;
+
+        internal LicenseFileText(string text, string licenseFileHeader, Func<string,Task<string>> loadFileFromPackage, string licenseFileLocation)
+        {
+            _text = text;
+            _licenseHeader = licenseFileHeader;
+            _licenseText = string.Format(CultureInfo.CurrentCulture, Resources.LicenseFile_Loading);
+            _loadFileFromPackage = loadFileFromPackage;
+            _licenseFileLocation = licenseFileLocation;
+        }
+
+        internal Lazy<object> LoadLicenseFile()
```

Comment:
This looks weird. Why make it return a `Lazy<object>`? Since it is a method and not a property is it even helpful to return a lazy which will always be `null` (because you are only returning `null`)?

Reply:
The caller in this case is the UI thread, a click event. 
This method kicks off a potentially heavy operation with JTF. 
The Lazy is there to prevent someone just clicking multiple times and us doing the same expensive operation twice. 
The null is returned there because there no such thing as Lazy<void>. 

Any other ideas are appreciated/happy to discuss offline.

Reply:
The lazy should've been a property, not a method. 

I've changed to interlocked because that's what we use in the SolutionRestoreWorker and throughout the rest of our code.